services/paymaster/src/constraint/tolls.rs

```diff
@@ -0,0 +1,100 @@
+use crate::rpc::ChainIndex;
+use borsh::BorshDeserialize;
+use fogo_sessions_sdk::tollbooth::TOLLBOOTH_PROGRAM_ID;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use serde_with::{serde_as, DisplayFromStr};
+use solana_program::instruction::CompiledInstruction;
+use solana_pubkey::Pubkey;
+use solana_transaction::versioned::VersionedTransaction;
+
+#[derive(Default, Deserialize, Clone)]
+#[serde_as]
+pub enum Tolls {
+    #[serde(rename = "free")]
+    #[default]
+    Free,
+    #[serde(rename = "fixed")]
+    Fixed(Vec<Toll>),
+}
+
+#[serde_as]
+#[derive(Deserialize, Clone)]
+
+pub struct Toll {
+    amount: u64,
+    #[serde_as(as = "DisplayFromStr")]
+    mint: Pubkey,
+}
+
+impl Tolls {
+    const MINT_ACCOUNT_INDEX_PAY_TOLL_INSTRUCTION: usize = 4;
+
+    pub async fn validate_toll_payment(
+        &self,
+        transaction: &VersionedTransaction,
+        chain_index: &ChainIndex,
+    ) -> Result<(), (StatusCode, String)> {
+        let tollbooth_instructions: Vec<(usize, &CompiledInstruction)> = transaction
```

Comment:
are we planning on releasing any SDK support for tollbooth? if so, could plug that in here to parse instead of manually checking